evalml/tests/demo_tests/test_datasets.py

```diff
@@ -86,5 +86,25 @@ def test_datasets(dataset_name, expected_shape, local_datasets):
 def test_datasets_match_local(dataset_name, demo_method, local_datasets):
     X, y = demo_method
     X_local, y_local = local_datasets[dataset_name]
+
+    if dataset_name == "daily_temp":
+        missing_date_1 = pd.DataFrame([pd.to_datetime("1984-12-31")], columns=["Date"])
+        missing_date_2 = pd.DataFrame([pd.to_datetime("1988-12-31")], columns=["Date"])
+        missing_y_1 = pd.Series([14.5], name="Temp")
+        missing_y_2 = pd.Series([14.5], name="Temp")
+
+        X_local = pd.concat(
```

Comment:
Just a suggestion, but what do you think about doing the `X_local` and `y_local` concats in 2 operations rather than 4? For instnace:
```python
X_local = pd.concat(
     [X_local.iloc[:1460], missing_date_1, X_local.iloc[1460: 2920], missing_date_2, X_local.iloc[2920:]]
).reset_index(drop=True)
```

Not a necessary change, but could be a little neater?

Reply:
Love it